src/lib/Persistence/Legacy/Notification/Gateway/DoctrineDatabase.php

```diff
@@ -151,12 +149,59 @@ public function loadUserNotifications(int $userId, int $offset = 0, int $limit =
         $query->orderBy(self::COLUMN_ID, 'DESC');
         $query->setParameter(':user_id', $userId, PDO::PARAM_INT);
 
-        return $query->execute()->fetchAll(PDO::FETCH_ASSOC);
+        return $query->execute()->fetchAllAssociative();
+    }
+
+    public function findUserNotifications(int $userId, ?NotificationQuery $query = null): array
+    {
+        $queryBuilder = $this->connection->createQueryBuilder();
+        $queryBuilder
+            ->select(...$this->getColumns())
+            ->from(self::TABLE_NOTIFICATION)
+            ->where($queryBuilder->expr()->eq(self::COLUMN_OWNER_ID, ':user_id'))
```

Comment:
Prefer `andWhere` over `where` (and especially `orWhere`), even if it's just one criterion.
```suggestion
            ->andWhere($queryBuilder->expr()->eq(self::COLUMN_OWNER_ID, ':user_id'))
```